### ISSUE TITLE
Copy from page to block blob types? 

### ISSUE THREAD

**Issue by thedavegray** [Azure/azure-storage-net-data-movement#180]
### Which service(blob, file) does this issue concern? 
page blobs -> block blobs 

 
### Which version of the SDK was used?
Microsoft.WindowsAzure.Storage.DataMovement 0.9.0

### On which platform were you using? (.Net Framework version or .Net Core version, and OS version)
.net 4.7.2 

### How can the problem be reproduced? It'd be better if the code caused the problem can be shared.
Copying from page to blob throws bot supported error.  

 
### What problem was encountered?
I see this has been addressed in the CLI version of AzCopy so I wondered if these libraries had been updated as well 
 
### Have you found a mitigation/solution?
page -> file -> block.

**Comment by EmmaZhu** [comment]
Hi @thedavegray 
By the CLI version AzCopy, do you mean AzCopy v10?

DMLib explicitly forbids copying between different blob types, but the internal logic supports to copy from page to block blob.  Is this behavior blocking your scenario? If there's requirement to copy between different blob types, we should consider to support it.

Thanks
Emma

**Comment by thedavegray** [comment]
Hi @EmmaZhu 

Yes I do mean AzCopy (v10) when I say CLI.  

I've got this (v10) working with page->block and it's speed is impressive.  Will this functionality be available in the SDK (Microsoft.WindowsAzure.Storage.DataMovement) library in the future? 

The reason we need this functionality is SQL backup to (url) Azure container (prior to 2016) creates page blobs as backup files.  We have 100s of terabytes of these and the only way to convert these to block blob,  and then make use of the archive tier, is copy blobs via an Azure file share.  This takes about a seven days of processor time each month.  

page->file->block 

Unfortunately upgrading the SQL instances to 2017/19 is not an option due to 3rd part dependencies. 

Thanks

**Comment by EmmaZhu** [comment]
Hi @thedavegray ,

Current DataMovement Libraray targets to REST API of "2018-11-09", and we won't release DataMovement Library with lower API version.

DataMovement Library forbids copying between different blob types at the very beginning of the transfer method. For DataMovement Library is an open sourced project, I think you can build your own library to remove this limitation.

You can remove the limitation by removing following code in file SingleObjectTransfer.cs
```
                if (sourceBlob.BlobType != destBlob.BlobType)
                {
                    throw new InvalidOperationException(Resources.SourceAndDestinationBlobTypeDifferent);
                }
```

If you'd want to target to API version 2017-11-09, DataMovement Library 0.8.0 should work for you: 

https://github.com/Azure/azure-storage-net-data-movement/releases/tag/v0.8.0 

Thanks
Emma

**Comment by thedavegray** [comment]
Hi, 
Are you saying you are not going to remove this limitation on the DM library?  That means it's not in step with the CLI version.  Sorry I don't understand, do I have to fork the existing code and create my own or can I download a new DLL to include in my own projects?  The link you sent just offers the source code, I don't want to compile or maintain my own I just want to add the assembly into my own projects. 

Basically I just want to use the latest version of your code in my own project, is this possible? 

Thanks

**Comment by EmmaZhu** [comment]
Hi @thedavegray ,

We are considering to loose the limitation in a future release. 

Would you only need to copy from page blob to block blob, or would you also need copying on other directions? Would the server side sync copy work for you? Server side sync copy will leverage REST API [PutBlockFromURL](https://docs.microsoft.com/en-us/rest/api/storageservices/put-block-from-url) .

Thanks
Emma
